src/Core/tests/WebTestCase.php

```diff
@@ -0,0 +1,288 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use function func_num_args;
+
+abstract class WebTestCase extends \Symfony\Bundle\FrameworkBundle\Test\WebTestCase
+{
+    protected KernelBrowser $client;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        static::createClient();
+        static::loadFixture(...static::getClassFixtures());
+    }
+
+    /** @return FixtureInterface[] */
+    protected static function getClassFixtures(): array
+    {
+        return [];
+    }
+
+    final protected static function loadFixture(FixtureInterface ...$fixture): void
+    {
+        (new ORMExecutor(self::getContainer()->get('doctrine.orm.entity_manager')))->execute($fixture, true);
+    }
+
+    /**
+     * @param string[] $expectedContent
+     * @param array<string, mixed> $requestParameters
+     */
+    final protected static function assertPageLoadedCorrectly(
+        string $url,
+        ?string $pageTitle = null,
+        array $expectedContent = [],
+        int $httpStatusCode = Response::HTTP_OK,
+        string $requestMethod = Request::METHOD_GET,
+        array $requestParameters = []
+    ): void {
+        static::assertHttpStatusCode($url, $httpStatusCode, $requestMethod, $requestParameters);
+        if ($pageTitle !== null) {
+            self::assertPageTitleSame($pageTitle, 'Page title is not "' . $pageTitle . '".');
+        }
+        static::assertResponseHasContent(...$expectedContent);
+    }
+
+    final protected static function assertHasLink(string $text, string $url): void
+    {
+        static::assertMinCount(
+            1,
+            static::getCrawler()->filter('a:contains("' . $text . '")[href="' . $url . '"]')
+        );
+    }
+
+    public static function assertMinCount(int $expectedCount, Countable|iterable $haystack, string $message = ''): void
+    {
+        static::assertGreaterThanOrEqual($expectedCount, count($haystack), $message);
+    }
+
+    public static function assertMaxCount(int $expectedCount, Countable|iterable $haystack, string $message = ''): void
+    {
+        static::assertLessThanOrEqual($expectedCount, $haystack, $message);
+    }
+
+    /**
+     * @param string[] $expectedContent
+     * @param array<string, mixed> $requestParameters
+     */
+    final protected static function assertClickOnLink(
+        string $linkText,
+        array $expectedContent,
+        int $httpStatusCode = Response::HTTP_OK,
+        string $requestMethod = Request::METHOD_GET,
+        array $requestParameters = []
+    ): void {
+        static::assertPageLoadedCorrectly(
+            static::getClient()->getCrawler()->selectLink($linkText)->link()->getUri(),
+            null,
+            $expectedContent,
+            $httpStatusCode,
+            $requestMethod,
+            $requestParameters
+        );
+    }
+
+    final protected static function assertResponseHasContent(string ...$content): void
```

Comment:
Rebased and fixed

Reply:
hmm, in PHPUnit itself the methods are without the `Has` part, `assertContains`, `assertStringContainsString`, `assertContainsOnly`, ... That was why I proposed `assertResponseContains` instead of `assertResponseHasContent`.

But it is naming, so it will always be a personal preference. So if you don't agree: fine by me.